src/characteristic/abstractCharacteristic.ts

```diff
@@ -134,4 +169,87 @@ export default abstract class AbstractCharacteristic<
       ...parameters,
     )
   }
+
+  deviceUsesFahrenheit(): boolean {
+    return this.device.getDevice().countryCode.startsWith('US')
+  }
+
+  roundHalf(r: number): number {
+    return Math.round(r * 2) / 2
+  }
+
+  /* 
+  LG Air Conditioners use a lookup table to go between Celsius and Farenheit, which we request from the 
+  LG servers as part of the GetDashboard call. There are two tables, CelToFah and FahToCel, which are *not* 
+  inverses of each other (unfortunately). 
+
+  These tables are the same as the units use, so if you want your results to match what they physically display,
+  you need to use the lookup tables. 
+
+  HomeKit's API uses Celsius as its standard unit, regardless of whether the user sees Farenheit on the UI or not--that is,
+  conversion happens on the iPhone. 
+
+  Thus, to ensure the temperatures on the app and the unit match, we have to convert to Farenheit, 
+  then back to Celsius for HomeKit, the latter conversion using the classic (5/9) - 32 math. 
+
+  We do all this only if the user sees Farenheit, as I *think* this issue is moot if the units show celsius? A non-US user to confirm.
+  */
+  getHomeKitCelsiusForLGAPICelsius(_celsius: number): number {
+    if (!this.deviceUsesFahrenheit()) {
+      return _celsius
```

Comment:
Is there harm in always performing the conversion? I believe both Celsius and Fahrenheit users will see these discrepancies due to rounding differences, right?

Reply:
See above